idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1144 @@
+                           doc="Packing channel size (m)")
+        
+        self.hydraulic_diameter = Expression(expr=4 * self.eps_ref / self.packing_specific_area,
+                                 doc="Hydraulic diameter (m)")
+
+        # Add the integer indices along vapor phase length domain
+        self.zi = Param(self.vapor_phase.length_domain, mutable=True,
+                        doc='''Integer indexing parameter required for transfer
+                             across boundaries of a given volume element''')
+                             
+        # Set the integer indices along vapor phase length domain
+        for i, x in enumerate(self.vapor_phase.length_domain, 1):
+            self.zi[x] = i
+
+        # Unit Model Design Variables
+        # Geometry
+        self.diameter_column = Var(domain=Reals,
+                                   initialize=0.1,
+                                   units=pyunits.m,
+                                   doc='Column diameter')
+        
+        self.area_column = Var(domain=Reals,
+                               initialize=0.5,
+                               units=pyunits.m**2,
+                               doc='Column cross-sectional area')
+        
+        self.length_column = Var(domain=Reals,
+                                 initialize=4.9,
+                                 units=pyunits.m,
+                                 doc='Column length')
+
+        # Hydrodynamics
+        self.velocity_vap = Var(self.flowsheet().time,
+                                self.vapor_phase.length_domain,
+                                domain=NonNegativeReals,
+                                initialize=2,
+                                units=pyunits.m / pyunits.s,
+                                doc='Vapor superficial velocity')
+        
+        self.velocity_liq = Var(self.flowsheet().time,
+                                self.liquid_phase.length_domain,
+                                domain=NonNegativeReals,
+                                initialize=0.01,
+                                units=pyunits.m / pyunits.s,
+                                doc='Liquid superficial velocity')
+        
+        self.holdup_liq = Var(self.flowsheet().time,
+                              self.liquid_phase.length_domain,
+                              initialize=0.001,
+                              doc='Volumetric liquid holdup [-]')
+
+        def rule_holdup_vap(blk, t, x):
+            return blk.eps_ref - blk.holdup_liq[t, x]
+
+        self.holdup_vap = Expression(self.flowsheet().time,
+                                     self.vapor_phase.length_domain,
+                                     rule=rule_holdup_vap,
+                                     doc='Volumetric vapor holdup [-]')
+        
+        # Calculating gas velocity at flooding point
+        def rule_flood_velocity(blk, t, x):
+            x_liq = blk.liquid_phase.length_domain.at(blk.zi[x].value)
```

Comment:
This correlation should have a doc string giving a reference , and maybe a little explanation (and note the specific units used).